ROX-26364: Release notes for 4.6 (#84701)
openshift/openshift-docs: release_notes/46-release-notes.adoc

:_mod-docs-content-type: ASSEMBLY
[id="release-notes-45"]
= Red{nbsp}Hat Advanced Cluster Security for Kubernetes 4.6
include::modules/common-attributes.adoc[]
:context: release-notes-46

toc::[]

{rh-rhacs-first} is an enterprise-ready, Kubernetes-native container security solution that protects your vital applications across the build, deploy, and runtime stages of the application lifecycle.
{product-title} deploys into your infrastructure and integrates with your DevOps tools and workflows. This integration provides better security and compliance, enabling DevOps and InfoSec teams to operationalize security.

.Release dates
[options="header"]
|====

|{product-title-short} version |Released on

|`4.6.0` | 2 December 2024

|====

[id="about-this-release-450_{context}"]
== About release 4.6.0

{product-title-short} 4.6 includes the following new features, improvements, and updates:

[id="new-features_{context}"]
== New features

This release adds improvements related to the following components and concepts:

[id="arm-support-secured-clusters_{context}"]

> === Support for ARM architecture in secured clusters

Reviewer: This is tech preview

We are adding support for ARM architecture in {product-title-short} secured clusters. This update allows users to leverage ARM’s efficient power consumption and high performance-per-watt benefits, making it ideal for resource-intensive tasks and cost-effective scaling, enhancing both flexibility and performance.

[id="scanner-v4-updates_{context}"]
=== Scanner V4 updates

* Scanner V4 uses VEX for vulnerability data: No release note text provided in Jira?
* Support for node scanning with Scanner V4 (Technical Preview): No release note text provided in Jira?

[id="policy-as-code_{context}"]
=== Configuring policies as code

No release note text in Jira?

[id="compliance-reporting-updates_{context}"]
=== Compliance reporting (Technology Preview)

> //not sure about this - fix version is 4.7, but label says 4.6

Reviewer: I updated the label to 4.6.0

* Sending the report periodically by email every time a scan is scheduled
* Creating email notifiers when creating a scan configuration as a destination to send on-demand reports
* Generating an on-demand report for a given scan configuration, which will be sent to any notifiers configured on that scan configuration

[id="network-graph-external-IP_{context}"]
== External IP address information available in network graph

The network graph now provides additional insights into external entities. With this update, you can view specific IP addresses associated with these external connections, offering a more comprehensive understanding of network activity.

[id="microsoft-sentinel-notifier_{context}"]
=== Microsoft Sentinel notifier added

{product-title-short} has added a Microsoft Sentinel notifier to send alerts and audit logs to Azure Log Analytics Workspace.

[id="s3-clone-support-backup_{context}"]
=== Support for backups using non-AWS S3 compatible providers

{product-title-short} has added a new external backup integration for non-AWS S3 compatible providers.

[id="violations-platform-components_{context}"]
=== Platform components displayed in the Violations page

> No release note text in Jira?

Reviewer: I have sent a reminder to Boaz in the PM channel

Reviewer: I think this tech preview. Boaz needs to confirm.

For the most recent list of major functionality deprecated and removed, see the following table.
Additional information about some removed or deprecated functionality is available after the table.

In the table, features are marked with the following statuses:

* GA: General Availability
* TP: Technology Preview
* DEP: Deprecated
* REM: Removed
* NA: Not applicable

.Deprecated and removed features tracker
[cols="4,1,1,1",options="header"]
|===
|Feature |{product-title-short} 4.4 |{product-title-short} 4.5 |{product-title-short} 4.6

| API token authentication for OpenShift Cluster Manager
| GA
| GA
| DEP

|`definitions.stackrox.io`
|DEP
|DEP
|

| Google Container Registry integration
| GA
| GA
| DEP

|Kernel support packages and driver download functionality
|NA
|DEP
|DEP

|Reporting of Istio vulnerabilities
|DEP
|DEP
|

| `rhacs-collector*` and `rhacs-collector-slim*` images
|NA
|DEP
|DEP

|`roxctl connectivity-map`
|DEP
|DEP
|

|`roxctl generate netpol`
|DEP
|DEP
|

|`stackrox-db` Central PVC
|GA
|GA
|REM

|`/v1/availableAuthProviders` endpoint
|GA
|DEP
|DEP

> |`/v1/clustercves/suppress` APIs

Reviewer 2: It is not completely removed but moved behind env var `ROX_VULN_MGMT_LEGACY_SNOOZE` which can be enabled if someone wants to continue using the feature. The env var is disabled by default. Same for other suppress and unsuppress endpoints for node and cluster.

|DEP
|DEP
|

|`/v1/clustercves/unsuppress` APIs
|DEP
|DEP
|

|`/v1/cve/requests` APIs
|DEP
|DEP
|

|`/v1/nodecves/suppress` APIs
|DEP
|DEP
|

|`/v1/nodecves/unsuppress` APIs
|DEP
|DEP
|

> |`/v1/summary/counts` endpoint

Reviewer 2: Not removed

|NA
|DEP
|DEP

|`/v1/tls-challenge` endpoint
|DEP
|DEP
|

> |Vulnerability Management (1.0) menu item

Reviewer 2: Not removed. Visible as Dashboard (deprecated) under VM

|DEP
|DEP
|

> |Vulnerability Report Creator permission

Reviewer 2: Not removed

:_mod-docs-content-type: ASSEMBLY

> [id="release-notes-45"]

Reviewer 3: 46?

= Red{nbsp}Hat Advanced Cluster Security for Kubernetes 4.6
include::modules/common-attributes.adoc[]
:context: release-notes-46

toc::[]

{rh-rhacs-first} is an enterprise-ready, Kubernetes-native container security solution that protects your vital applications across the build, deploy, and runtime stages of the application lifecycle.
{product-title} deploys into your infrastructure and integrates with your DevOps tools and workflows. This integration provides better security and compliance, enabling DevOps and InfoSec teams to operationalize security.

.Release dates
[options="header"]
|====

|{product-title-short} version |Released on

|`4.6.0` | 2 December 2024

|====

[id="about-this-release-450_{context}"]
== About release 4.6.0

{product-title-short} 4.6 includes the following new features, improvements, and updates:

[id="new-features_{context}"]
== New features

This release adds improvements related to the following components and concepts:

[id="arm-support-secured-clusters_{context}"]
=== Support for ARM architecture in secured clusters (Technology Preview)

:FeatureName: Support for ARM architecture in secured clusters
include::snippets/technology-preview.adoc[]

> We are adding support for ARM architecture in {product-title-short} secured clusters. This update allows users to leverage the efficient power consumption and high performance-per-watt benefits of ARM, making it ideal for resource-intensive tasks and cost-effective scaling and enhancing both flexibility and performance.

Reviewer 3: Do you happen to know if we support it for Central clusters, yet? If not, should we make it clear it is still not supported there? I honestly don't know who to ask

For more information, see **LINK HERE**.

[id="scanner-v4-updates_{context}"]
=== Scanner V4 now uses CSAF/VEX for vulnerability data

> The vulnerability reporting for Red{nbsp}Hat content in {product-title-short} vulnerability management is changing with the introduction of Common Security Advisory Framework (CSAF) Vulnerability Exploitability eXchange (VEX) vulnerability data as a security data source. Red{nbsp}Hat is switching to CSAF VEX as a standardized security advisory format to communicate vulnerabilities affecting Red{nbsp}Hat products. In particular, the  VEX profile in CSAF format is now used to describe, with greater detail than before, which Red Hat products and components are impacted or known not to be impacted by a specific vulnerability identified by the Common Vulnerability and Exposures (CVE) ID.

Reviewer 3: We are only using VEX files, and not CSAF. Sorry for the confusion (I know that's how we've been saying it and how the tickets are written). I think this is more accurate:

```suggestion
The vulnerability reporting for Red{nbsp}Hat content in {product-title-short} vulnerability management is changing with the introduction of Red{nbsp}Hat Common Security Advisory Framework (CSAF) Vulnerability Exploitability eXchange (VEX) vulnerability data as a security data source. Red{nbsp}Hat is switching to CSAF-VEX as standardized security vulnerability formats to communicate vulnerabilities affecting Red{nbsp}Hat products[IF YOU WANT TO ADD A LINK, THIS MAY BE GOOD EVEN THOUGH IT'S OLD AND FROM A TIME BEFORE RED HAT HAD VEX DATA: https://www.redhat.com/en/blog/future-red-hat-security-data or https://www.redhat.com/en/blog/vulnerability-exploitability-exchange-vex-beta-files-now-available or https://www.redhat.com/en/blog/red-hat-vex-files-cves-are-now-generally-available (maybe even all three)]. In particular, the VEX profile is now used to describe, with greater detail than before, which Red Hat products and components are impacted or known not to be impacted by a specific vulnerability identified by the Common Vulnerability and Exposures (CVE) ID.
```

> CSAF VEX is available when using Scanner V4 for vulnerability scanning.

Reviewer 3: From the blog posts I linked above I think it's `CSAF-VEX` but honestly I have no clue. I'm just trying to use context clues but I could be wrong

[id="microsoft-sentinel-notifier_{context}"]
=== Microsoft Sentinel notifier added

{product-title-short} has added a Microsoft Sentinel notifier to send alerts and audit logs to Azure Log Analytics Workspace. For more information, see xref:../integration/integrating-with-microsoft-sentinel-notifier.adoc#integrating-with-microsoft-sentinel-notifier[Integrating with Microsoft Sentinel notifier].

[id="s3-clone-support-backup_{context}"]
=== Support for backups using non-AWS S3 compatible providers

{product-title-short} has added a new external backup integration for non-AWS S3 compatible providers. For more information, see xref:../integration/integrate-with-s3-api-compatible-services.adoc#integrate-with-s3-api-compatible-services[Integrating with S3 API compatible services].

[id="violations-platform-components_{context}"]
=== Platform components displayed in the Violations page

No release note text in Jira?

[id="vulnerability-management-updates_{context}"]
=== Vulnerability Management page updates

CVE published date now included::

> {product-title-short}, in vulnerability management, now reports the CVE published date. The CVE published date is the date when the CVE entry was first published in the National Vulnerability Database (NVD) or by the vendor in vendor specific security data feeds when those are available.

Reviewer 3: I think it does, but I want to ensure it is clear we prefer the date from the vendor and then only use NVD when that info is missing

* The endpoint `/v2/compliance/scan/configurations/reports/run` method has changed from `PUT` to `POST`.

[id="documentation-updates-450_{context}"]
== Documentation updates

[id="deprecated-and-removed-features_{context}"]
== Deprecated and removed features

Some features available in earlier releases have been deprecated or removed.

Deprecated functionality is still included in {product-title-short} and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments.
For the most recent list of major functionality deprecated and removed, see the following table.
Additional information about some removed or deprecated functionality is available after the table.

In the table, features are marked with the following statuses:

* GA: General Availability
* TP: Technology Preview
* DEP: Deprecated
* REM: Removed
* NA: Not applicable

.Deprecated and removed features tracker
[cols="4,1,1,1",options="header"]
|===
|Feature |{product-title-short} 4.4 |{product-title-short} 4.5 |{product-title-short} 4.6

| API token authentication for Red Hat OpenShift Cluster Manager^[1]^
| GA
| GA
| DEP

> |`definitions.stackrox.io`

Reviewer 3: we deprecated this?

Author: I posted the RN link and PR for it on slack.

Reviewer 3: guess we'll keep it deprecated, but not sure if we plan on doing anything about this... 🤷

|DEP
|DEP
|

| Google Container Registry integration^[2]^
| GA
| GA
| DEP

|Kernel support packages and driver download functionality ^[3]^
|NA
|DEP
|DEP

> |Reporting of Istio vulnerabilities

Reviewer 3: Still deprecated, not removed from what I can tell

|DEP
|DEP
|

> | `rhacs-collector*` and `rhacs-collector-slim*` images^[4]^

Reviewer 3: `rhacs-collector` is deprecated?

Reviewer 3: I believe this has been resolved in 4.5 docs? From what I can tell, the conclusion is this should just say `rhacs-collector-slim*` is deprecated

Reviewer 3: I believe we still need to remove `rhacs-collector*` from this

|NA
|DEP
|DEP

|`roxctl connectivity-map`
|DEP
|DEP
|

|`roxctl generate netpol`
|DEP
|DEP
|

|`stackrox-db` Central PVC^[5]
|GA
|GA
|REM

|`/v1/availableAuthProviders` endpoint
|GA
|DEP
|DEP

|`/v1/clustercves/suppress` APIs^[6,7]^
|DEP
|DEP
|DEP

|`/v1/clustercves/unsuppress` APIs^[6,7]^
|DEP
|DEP
|DEP

|`/v1/cve/requests` APIs^[8]^
|DEP
|DEP
|REM

|`/v1/nodecves/suppress` APIs^[6,7]^
|DEP
|DEP
|DEP

|`/v1/nodecves/unsuppress` APIs^[6,7]^
|DEP
|DEP
|DEP

|`/v1/summary/counts` endpoint
|NA
|DEP
|DEP

|`/v1/tls-challenge` endpoint
|DEP
|DEP
|

|Vulnerability Management (1.0) menu item^[9]^
|DEP
|DEP
|DEP

|Vulnerability Report Creator permission
|DEP
|DEP
|DEP

|===

. API token authentication has been deprecated by Red Hat OpenShift Cluster Manager. The corresponding cloud source integration now uses service accounts for authentication.

> . The Google Container Registry integration is now deprecated. Users should use Artifact Registry as a registry replacement and Scanner V4 as a scanner replacement.

Reviewer 3: If you want a link - https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

. Kernel support packages and driver download functionality are deprecated.

> . The `rhacs-collector*` and `rhacs-collector-slim*` images have been deprecated. They are now functionally the same and do not include kernel drivers.

Reviewer 3: same as above. I believe this has been resolved in 4.5 docs and `rhacs-collector` is not deprecated

Hiding unwanted display columns::

{product-title-short} now provides the ability to hide unwanted columns in tables by using column management. Starting with the Workload CVEs section of {product-title-short}, a button is displayed above the table containing the text "Columns" and provides a count of the number of columns that are enabled. You can click this button to open a modal and hide unwanted columns to reduce visual noise. These settings are saved per table in your browser and will remain in place across separate sessions.

[id="certifications_{context}"]
== Certifications

{product-title-managed} is certified across key global standards ensuring top-tier security, compliance, and data protection for your business, including the following standards:

* ISO/IEC 27001:2022
* ISO/IEC 27017:2015
* ISO/IEC 27018:2019
* PCI DSS 4.0
* SOC 2 Type 2
* SOC 2 Type 3

[id="notable-technical-changes_{context}"]
== Notable technical changes

This release contains the following changes:

> `release stage` API property:: The `release stage` property has been added to the `/v1/featureflags` response in the {product-title-short} API. When querying the API, it returns a list of features, the name of the environment variable for each feature, and whether or not the feature is enabled. Some features might be enabled or disabled if they are released as a Technology Preview. For more information, see xref:../configuration/managing-preview-features.adoc#managing-feature-flags_managing-preview-features[Managing feature flags].

Reviewer 4: No such 'release stage' property, we dropped the feature.

.Release dates
[options="header"]
|====

|{product-title-short} version |Released on

|`4.6.0` | 2 December 2024

|====

[id="about-this-release-450_{context}"]
== About release 4.6.0

{product-title-short} 4.6 includes the following new features, improvements, and updates:

Architecture::

* xref:../release_notes/46-release-notes.adoc#arm-support-secured-clusters_release-notes-46[Support for ARM architecture in secured clusters (Technology Preview)]

Compliance::

* xref:../release_notes/46-release-notes.adoc#certifications_release-notes-46[Certifications for {product-title-managed}]
* xref:../release_notes/46-release-notes.adoc#compliance-reporting-updates_release-notes-46[Compliance reporting (Technology Preview)]

Documentation::

* xref:../release_notes/46-release-notes.adoc#api-docs_release-notes-46[API documentation available publicly]

Network::

* xref:../release_notes/46-release-notes.adoc#network-graph-external-IP_release-notes-46[External IP address information available in network graph]

Platform::

> * xref:../release_notes/46-release-notes.adoc#microsoft-sentinel-notifier_release-notes-46[Microsoft Sentinel notifier]

Reviewer 5: Should this highlight more that the Sentinel Notifier is a new feature? For example:

```suggestion
* xref:../release_notes/46-release-notes.adoc#microsoft-sentinel-notifier_release-notes-46[Added Microsoft Sentinel notifier]
```

. The `/v1/cve/requests` API for managing vulnerability exceptions is removed. Use the new `/v2/vulnerability-exceptions/` API.
. The *Dashboard* view under *Vulnerability Management* is deprecated. Use the *Workload CVEs*, *Exception Management*, *Platform CVEs*, and *Node CVEs* views as alternatives.

[id="deprecated-features_{context}"]
=== Deprecated features

The following section provides information about additional deprecated features:

* To unify the response data for stream and unary API requests:
** The `error` field returned for failed unary API requests is deprecated. Instead of the `error` field, use the `message` field to retrieve error information. The `message` field has the same information as the `error` field.
//is this deprecated in 4.6 or removed?

> ** In this release, Red{nbsp}Hat removed the `grpcCode`, `httpCode`, and `httpStatus` fields in returned error response for gRPC stream APIs. Instead, the response includes a new field, `code` which includes the `grpcCode` data.

Reviewer 6: ```suggestion
** In this release, Red{nbsp}Hat removed the following fields in the response for gRPC stream APIs:
*** `grpcCode`
***  `httpCode`
*** `httpStatus` 
With this release, the response includes the new `code` field that includes the `grpcCode` data.
```
WDYT?

//the 4.5 notes say that these above will be removed in the next release. please confirm.

[id="bug-fixes_{context}"]
== Bug fixes in version 4.6.0

*Release date*: 2 December 2024

* Fixed incorrect timestamp data for the *First discovered* column on the *Workload CVE* single page when viewing affected images.

* Fixed incorrect CVE counts when viewing an image in the *Vulnerability Management* window that contains CVEs with an unknown severity.

* In runtime monitoring, process names and arguments could cause serialization problems when containing invalid UTF-8 characters. This resulted in error messages in the collector logs. Those characters are now filtered and replaced with a `?` when necessary.

> * Fixed a bug when using delegated scanning where newer image metadata and layers were pulled incorrectly for an older image referenced by tag when the image registry contents have changed since deployment. Now, the metadata and layers pulled are based on the digest of the image provided by the container runtime when available, instead of just the tag.

Reviewer 6: I think that these would be easier to understand if they followed the default bug fix format, similar to this example: https://docs.google.com/presentation/d/1IRuI_CMcMIgQtjfGG2o1PWugMFv_B_bYCQ0rPUuP40c/edit#slide=id.gb820dbee29_1_49

So, something like: 
* Before this release, the timestamp data displayed incorrectly when viewing affected images in the *First discovered* column of the *Workload CVE* single page view. This update resolves the issue.
